js/base/Exchange.js

```diff
@@ -1634,6 +1634,71 @@ module.exports = class Exchange {
         }
     }
 
+    networkCodeToId (networkCode) {
+        const networks = this.safeValue (this.options, 'networks', {});
+        return this.safeString (networks, networkCode, networkCode);
+    }
+
+    networkIdToCode (networkId) {
+        const networksById = this.safeValue (this.options, 'networksById', {});
+        return this.safeString (networksById, networkId, networkId);
+    }
+
+    handleNetworkCodeAndParams (params) {
+        const networkCodeInParams = this.safeString2 (params, 'networkCode', 'network');
+        if (networkCodeInParams !== undefined) {
+            params = this.omit (params, [ 'networkCode', 'network' ]);
+        }
+        // if it was not defined by user, we should not set it from 'defaultNetworks', because handleNetworkCodeAndParams is for only request-side and thus we do not fill it with anything. We can only use 'defaultNetworks' after parsing response-side
+        return [ networkCodeInParams, params ];
+    }
+
+    defaultNetworkCode (currencyCode) {
+        let defaultNetworkCode = undefined;
+        const defaultNetworks = this.safeValue (this.options, 'defaultNetworks', {});
+        if (currencyCode in defaultNetworks) {
+            // if currency had set its network in "defaultNetworks", use it
+            defaultNetworkCode = defaultNetworks[currencyCode];
+        } else {
+            // otherwise, try to use the global-scope 'defaultNetwork' value (even if that network is not supported by currency, it doesn't make any problem, this will be just used "at first" if currency supports this network at all)
+            const defaultNetwork = this.safeValue (this.options, 'defaultNetwork');
+            if (defaultNetwork !== undefined) {
+                defaultNetworkCode = defaultNetwork;
+            }
+        }
+        return defaultNetworkCode;
+    }
+
+    getDefaultNetworkIdFromRawNetworks (currencyCode, networkCode, networkEntriesIndexed) {
```

Comment:
@ttodua  "Default" does not seem correct here because if `networkCode` is specified, it might not be the default.

What do you think of `getNetworkIdFromAvailableNetworks` ?

Reply:
You are somewhat right, I've changed around 5 different names in the process frankly, and none of them seemed absolutely fit. on one hand,  if `networkCode` is provided, then it ignores any "defaults" or whatever, but if `networkCode` is not passed, then it yes, tries to detect 'default network'. but in overall, i think your version will be less intrusive. I'm pushing it now.

Reply:
done. tests passes now, `bitbns` no longer breaks the build.